### DISCUSSION TITLE
Forcing Launcher update

### DISCUSSION THREAD

**Discussion by Bartixxx32** [SKCraft/Launcher#464] [General]
Hello, i have question, is there posibility to force launcher to get updated, instead of user click Update button?

**Comment by hedgehog1029** [comment]
There isn't a default method of forcing launcher updates. There's a couple of useful features:

1. The launcher passes its current version as a parameter when fetching the news URL; you can use some server-side logic to add a message asking users to update their launcher if you detect an outdated one.
2. If you really need to, you can change the protocol version of the launcher & your manifests, which will force the user to update the launcher before they can update the packs. However this should really only be used if you *do* have breaking manifest changes that will break old launchers.

**Reply by Bartixxx32** [reply]
That, was a quick response! i`ll look at this 1st method then, thanks !

**Reply by hedgehog1029** [reply]
If you look at [the example properties file](https://github.com/SKCraft/Launcher/blob/master/launcher/src/main/resources/com/skcraft/launcher/launcher.properties#L24) you can see that there's a `?version=%s` at the end - you'll need to make sure you've got that in order to pass the version on. (the `%s` will be replaced with the local launcher version)
